src/chainstate/burn/db/sortdb.rs

```diff
@@ -2862,7 +2854,7 @@ impl SortitionDB {
         conn: &Connection,
         consensus_hash: &ConsensusHash,
     ) -> Result<Option<SortitionId>, db_error> {
-        let qry = "SELECT sortition_id FROM snapshots WHERE consensus_hash = ?1";
+        let qry = "SELECT sortition_id FROM snapshots WHERE consensus_hash = ?1 AND pox_valid = 1 LIMIT 1";
```

Comment:
Are we sure that this changes are not breaking consensus?
It would impact the vectors returned by `process_new_block_arrivals`, which are then written to a MARF?

Reply:
The sortition DB's MARF isn't part of the consensus code.

Reply:
Also, block arrivals are going to be different for each node already, since nodes don't all download and process blocks in the same order.

Reply:
I see. So what are the properties that we're getting out of the MARF? Why not using a Hashmap?

Reply:
We need to keep track of which Stacks blocks have arrived on which burnchain forks in order to determine which Stacks fork is the longest Stacks fork on each burnchain fork's PoX-valid sortitions.  The canonical Stacks fork is not the longest Stacks fork anywhere; it's the longest Stacks fork on the PoX-valid sortitions on the longest burnchain.